crates/epoch-snark/src/gadgets/mod.rs

```diff
@@ -69,45 +73,36 @@ pub(super) fn pack<F: PrimeField, P: FpParameters>(
         .collect::<Result<Vec<_>, _>>()
 }
 
-fn to_fr<T: Into<u64>, CS: ConstraintSystem<Fr>>(
-    cs: &mut CS,
-    num: Option<T>,
-) -> Result<FrGadget, SynthesisError> {
-    FrGadget::alloc(cs, || Ok(Fr::from(num.get()?.into())))
-}
-
-fn fr_to_bits<CS: ConstraintSystem<Fr>>(
-    cs: &mut CS,
-    input: &FrGadget,
+fn fr_to_bits(
+    input: &FrVar,
     length: usize,
-) -> Result<Vec<Boolean>, SynthesisError> {
-    let mut input = input.to_bits(cs.ns(|| "input to bits"))?;
-    input.reverse();
+) -> Result<Vec<Bool>, SynthesisError> {
+    let input = input.to_bits_le()?;
     Ok(input[0..length].to_vec())
 }
 
-fn g2_to_bits<CS: ConstraintSystem<Fr>>(
-    cs: &mut CS,
-    input: &G2Gadget,
-) -> Result<Vec<Boolean>, SynthesisError> {
-    let x_0 = input.x.c0.to_bits(cs.ns(|| "aggregated pub key c0 bits"))?;
-    let x_1 = input.x.c1.to_bits(cs.ns(|| "aggregated pub key c1 bits"))?;
-    let y_bit =
-        YToBitGadget::<Parameters>::y_to_bit_g2(cs.ns(|| "aggregated pub key y bit"), &input)?;
+fn g2_to_bits(
+    input: &G2Var,
+) -> Result<Vec<Bool>, SynthesisError> {
+    let mut x_0 = input.x.c0.to_bits_le()?;
+    let mut x_1 = input.x.c1.to_bits_le()?;
+    x_0.reverse();
```

Comment:
Maybe a comment that we need big endian later on and why?

Reply:
Added a comment that the function returns big-endian elements. I think once to_bits_be is available we should reevaluate how to do bit conversions in a principled manner. I'm not sure there is a crisp explanation for why this function returns elements in BE vs. LE

Reply:
Sounds good. I think the best explanations is that that's how it matches our native implementation... But that's not motivation, just an excuse :)